lib/puppet/provider/package/homebrew.rb

```diff
@@ -8,18 +8,20 @@
   def install
     name = install_name
 
-    Puppet.debug "Installing #{name}"
-    output = execute([command(:brew), :install, name, *install_options])
-
-    if output =~ /Searching taps/
-      Puppet.debug "Falling back to brew-cask (still installing #{name}"
-      output = execute([command(:brew), :cask, :install, name, *install_options])
-      # brewcask includes some funky beer characters that f*ck with encoding
-      output = output.encode('UTF-8', :invalid => :replace, :undef => :replace)
-
+    begin
+      Puppet.debug "Looking for #{name} package on brew..."
+      output = execute([command(:brew), :info, name])
       if output.empty?
-        raise Puppet::ExecutionFailure, "Could not find package #{name}"
+         Puppet.debug "Package #{name} not found on Brew. Trying BrewCask..."
+         output = execute([command(:brew), :cask, :info, name], failonfail: true)
+         Puppet.debug "Package found on brewcask, installing..."
+         output = execute([command(:brew), :cask, :install, name, *install_options])
+      else
+        Puppet.debug "Package found, installing..."
+        output = execute([command(:brew), :install, name, *install_options], failonfail: true)
       end
+    rescue Puppet::ExecutionFailure => detail
+      raise Puppet::Error, "Could not find package: #{name}"
     end
 
     if output =~ /sha256 checksum/
```

Comment:
See comments above about checking to be sure we don't need to move this into the rescue block.